### PR TITLE
validate empty labells

### PR DESCRIPTION
**Pull request by jamesgorrie**
Picked up from a Sentry error.
* Validate the label input.
* I remove the cancel on blur as it cancels when you leave the input, including when you try to click the buttons
* changed naming / structure to match other component

cc: @tsop14 

label-validation

### CODE REVIEW

File: kahuna/public/js/components/gr-add-label/gr-add-label.html [hunk 1]
```diff
@@ -3,19 +3,36 @@
...
-            Add<span ng:show="ctrl.adding">ing...</span>
-        </span>
+    <span ng:hide="ctrl.grSmall">
+        Add<span ng:show="ctrl.adding">ing...</span>
+    </span>
 </button>
-<form class="gr-add-label__form" ng:if="ctrl.active" ng:submit="ctrl.addLabel()" >
-    <input class="gr-add-label__form__input" type='text' ng:model="ctrl.newLabel" gr:auto-focus placeholder="label1, label2, label3…" ng:blur="ctrl.cancel()"></input>
+<form
+    class="gr-add-label__form"
```
Comment (theefer) [inline kahuna/public/js/components/gr-add-label/gr-add-label.html]: Is this a new syntax for aligning attributes we want to introduce?

**Comment by theefer** [comment]
:+1:

File: kahuna/public/js/components/gr-add-label/gr-add-label.html (hunk 1 above)
Comment (jamesgorrie) [inline kahuna/public/js/components/gr-add-label/gr-add-label.html]: I've been doing it like for ages... just realised there is the other way too - the other way doesn't play nice with IntelliJ. How's it with ~Vim~ emacs?

**Comment by tsop14** [comment]
Noted. Thanks. :smile:

File: kahuna/public/js/components/gr-add-label/gr-add-label.html (hunk 1 above)
Comment (theefer) [inline kahuna/public/js/components/gr-add-label/gr-add-label.html]: GNU/Emacs:tm: handles both fairly well. I tend to prefer the way we've done it so far for readability reasons, but if it's a pain for all but me to edit those, and the format above works better for all, I don't mind switching to that.

**Pull request merged**